src/pkcs7.c

```diff
@@ -881,34 +881,18 @@ int get_attribute(STACK_OF(X509_ATTRIBUTE) *attribs, int required_nid,
 				ASN1_TYPE **asn1_type) {
```

Comment:
Instead of fixing this self build funktion I would recommend to use the function that comes with OpenSSL ( PKCS7_get_signed_attribute(si, nid) )

So the codeblock starting from line 841 would look like this:
```
asn1_type = PKCS7_get_signed_attribute(si, nid);
if (ASN1_TYPE_get(asn1_type) != type) {
    fprintf(stderr, "%s: wrong ASN.1 type\n",pname);    
    exit (SCEP_PKISTATUS_P7);
}
```